File: benchmarks/corpus-a-ground-truth/js/probe.js
const crypto = require('crypto');
const bcrypt = require('bcrypt');
const argon2 = require('argon2');

async function probe() {
  crypto.generateKeyPairSync('rsa', { modulusLength: 2048 });            // EXPECT rsa
  crypto.generateKeyPairSync('ec', { namedCurve: 'P-256' });             // EXPECT ecdsa
  crypto.createECDH('prime256v1');                                       // EXPECT ecdh
  crypto.generateKeyPairSync('dsa', { modulusLength: 2048 });            // EXPECT dsa
  crypto.createHash('md5');                                              // EXPECT md5
  crypto.createHash('sha1');                                             // EXPECT sha1
  crypto.createHash('sha256');                                           // EXPECT sha256
  crypto.createHash('sha384');                                           // EXPECT sha384
  crypto.createHmac('sha256', 'k');                                      // EXPECT hmac
  crypto.pbkdf2Sync('pw', 's', 600000, 32, 'sha256');                    // EXPECT pbkdf2
  crypto.scryptSync('pw', 's', 32);                                      // EXPECT scrypt
  await bcrypt.hash('pw', 10);                                           // EXPECT bcrypt
  await argon2.hash('pw');                                               // EXPECT argon2
  crypto.createCipheriv('aes-128-gcm', k, iv);                           // EXPECT aes128
  crypto.createCipheriv('des-ede3-cbc', k, iv);                          // EXPECT 3des
  crypto.createCipheriv('rc4', k, null);                                 // EXPECT rc4
  crypto.createCipheriv('chacha20-poly1305', k, iv);                     // EXPECT chacha20
  await crypto.subtle.generateKey({ name: 'ML-KEM-768' }, true, ['encrypt']);  // EXPECT mlkem
  await crypto.subtle.generateKey({ name: 'ML-DSA-65' }, true, ['sign']);      // EXPECT mldsa
}

File: benchmarks/corpus-a-ground-truth/java/Probe.java
import java.security.*;
import javax.crypto.*;
import javax.crypto.spec.*;
import org.mindrot.jbcrypt.BCrypt;

public class Probe {
    void probe() throws Exception {
        KeyPairGenerator.getInstance("RSA").initialize(2048);              // EXPECT rsa
        KeyPairGenerator.getInstance("EC");                                // EXPECT ecdsa
        KeyAgreement.getInstance("ECDH");                                  // EXPECT ecdh
        KeyPairGenerator.getInstance("DSA");                               // EXPECT dsa
        MessageDigest.getInstance("MD5");                                  // EXPECT md5
        MessageDigest.getInstance("SHA-1");                                // EXPECT sha1
        MessageDigest.getInstance("SHA-256");                              // EXPECT sha256
        MessageDigest.getInstance("SHA-384");                              // EXPECT sha384
        Mac.getInstance("HmacSHA256");                                     // EXPECT hmac
        SecretKeyFactory.getInstance("PBKDF2WithHmacSHA256");              // EXPECT pbkdf2
        BCrypt.hashpw("pw", BCrypt.gensalt());                             // EXPECT bcrypt
        Cipher.getInstance("AES/GCM/NoPadding");                           // EXPECT aesgcm
        Cipher.getInstance("DESede/CBC/PKCS5Padding");                      // EXPECT 3des
        Cipher.getInstance("RC4");                                         // EXPECT rc4
        Cipher.getInstance("ChaCha20-Poly1305");                           // EXPECT chacha20
        KeyPairGenerator.getInstance("ML-KEM");                            // EXPECT mlkem
        KeyPairGenerator.getInstance("ML-DSA");                            // EXPECT mldsa
        KeyPairGenerator.getInstance("SLH-DSA");                           // EXPECT slhdsa
    }
}
